x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/SetupPasswordTool.java

```diff
@@ -248,10 +249,19 @@ public void close() {
             }
         }
 
-        void setupOptions(OptionSet options, Environment env) throws Exception {
+        void setupOptions(Terminal terminal, OptionSet options, Environment env) throws Exception {
             keyStoreWrapper = keyStoreFunction.apply(env);
-            // TODO: We currently do not support keystore passwords
-            keyStoreWrapper.decrypt(new char[0]);
+            try (SecureString keystorePassword = keyStoreWrapper.hasPassword() ?
```

Comment:
Since we don't need to write the keystore back out in these tools, can we have a utility method somewhere to read the keystore, instead of copying this code to each tool?